src/components/LoginForm/LoginForm.js

```diff
@@ -45,8 +45,11 @@ function findRoom(rooms, roomId) {
  * @param {Function} dispatch - the dispatch function available in the Redux store
  * @returns {Function}
  */
-function onSubmit(dispatch) {
+function onSubmit(dispatch, settings) {
   return function(data) {
+    settings.username = data.username;
+    settings.roomId = data.room;
+    dispatch(roomActions.saveSettings(settings));
     dispatch(roomActions.login(data.username, data.room, data.pin));
```

Comment:
JFYI: I have learned that React Redux added [`batch()`](https://react-redux.js.org/api/batch) to its API in version 7, which 

>  allows any React updates in an event loop tick to be batched together into a single render pas

However, it is useful only when using React < 18. Hopefully, we could upgrade the project to the latest React version soon.

Reply:
Thanks for the info ;)